express/src/services/lambda-facade/LambdaFacade.ts

```diff
@@ -3,78 +3,59 @@ import {QueryCommandOutput} from "@aws-sdk/client-dynamodb";
 import axios, {Axios, AxiosResponse} from "axios";
 import {OnboardingTableItem} from "../../../@types/OnboardingTableItem";
 import {Service} from "../../../@types/Service";
+import {lambda} from "../../config/environment";
 import AuthenticationResultParser from "../../lib/authentication-result-parser";
-import LambdaFacadeInterface from "./LambdaFacadeInterface";
+import LambdaFacadeInterface, {ClientUpdates, UserUpdates} from "./LambdaFacadeInterface";
 
 /* eslint-disable @typescript-eslint/no-unused-vars */
 class LambdaFacade implements LambdaFacadeInterface {
-    private instance: Axios;
+    private readonly client: Axios;
 
-    constructor(baseURL: string) {
-        this.instance = axios.create({
-            baseURL: baseURL,
+    constructor() {
+        console.log("Creating Lambda facade...");
+
+        this.client = axios.create({
+            baseURL: lambda.apiBaseUrl,
             headers: {
                 "Content-Type": "application/json"
             }
         });
     }
 
-    async putUser(user: OnboardingTableItem, accessToken: string): Promise<AxiosResponse> {
-        return await (
-            await this.instance
-        ).post("/put-user", user, {
-            headers: {
-                "authorised-by": accessToken
-            }
-        });
+    async putUser(user: OnboardingTableItem, accessToken: string): Promise<void> {
+        await this.post("/put-user", user, accessToken);
     }
 
-    async getUserByCognitoId(cognitoId: string, accessToken: string): Promise<AxiosResponse> {
-        return await (
-            await this.instance
-        ).post("/get-user", `user#${cognitoId}`, {
-            headers: {
-                "authorised-by": accessToken
-            }
-        });
+    getUserByCognitoId(cognitoId: string, accessToken: string): Promise<AxiosResponse> {
+        return this.post("/get-user", `user#${cognitoId}`, accessToken);
     }
 
-    async newService(service: Service, userId: string, email: string, accessToken: string): Promise<AxiosResponse> {
+    async newService(service: Service, userId: string, email: string, accessToken: string): Promise<void> {
         const body = {
             service: service,
             userDynamoId: userId,
             userEmail: email
         };
-        return await (
-            await this.instance
-        ).post("/new-service", JSON.stringify(body), {
-            headers: {
-                "authorised-by": accessToken
-            }
-        });
+
+        await this.post("/new-service", JSON.stringify(body), accessToken);
     }
 
-    async generateClient(service: Service, authenticationResult: AuthenticationResultType): Promise<AxiosResponse> {
+    generateClient(service: Service, authenticationResult: AuthenticationResultType): Promise<AxiosResponse> {
```

Comment:
I don't normally like storycreep but this would have been a good place to slot in the API key stuff.

Reply:
Will do this separately as this was already a big PR